pommesdispatch/model_funcs/helpers.py

```diff
@@ -120,7 +118,80 @@ def convert_annual_limit(annual_limit, start_time, end_time):
             new_limit += annual_limit
 
         # Add limits for fractions of the start year and end year
-        new_limit += (annual_limit * start_year_fraction
-                      + annual_limit * end_year_fraction)
+        new_limit += (
+            annual_limit * start_year_fraction
+            + annual_limit * end_year_fraction
+        )
 
     return new_limit
+
+
+def convert_annual_costs_nominal_to_real(
+    nominal_costs, inflation_rate=1.02, year=2022
+):
+    """Convert cost values of DataFrame from nominal to real terms
+
+    Parameters
+    ----------
+    nominal_costs: :obj:`pd.DataFrame`
+        Nominal costs data in annual resolution (years = columns)
+
+    inflation_rate: :obj:`float`
+        Inflation rate
+
+    year: :obj:`int`
+        Year for which the nominal costs shall be expressed
+
+    Returns
+    -------
+    real_costs: :obj:`pd.DataFrame`
+        Real costs data in annual resolution
+    """
+    real_costs = nominal_costs.copy()
+    for column in real_costs.columns:
+        if column != "label":
+            try:
+                real_costs[column] = real_costs[column].div(
+                    inflation_rate ** (int(column) - year)
+                )
+            except TypeError:
+                msg = (
+                    "DataFrame format not as expected\n"
+                    "Except for column 'label', "
+                    "all other columns must be integer"
+                )
+                raise TypeError(msg)
+
+    return real_costs
+
+
+def convert_nominal_to_real_time_series(
+    nominal_time_series, inflation_rate=1.02, year=2022
+):
+    """Convert time series values of DataFrame from nominal to real terms
+
+    Parameters
+    ----------
+    nominal_time_series: :obj:`pd.DataFrame`
+        Nominal time series data in hourly resolution
+
+    inflation_rate: :obj:`float`
+        Inflation rate
+
+    year: :obj:`int`
+        Year for which the nominal costs shall be expressed
+
+    Returns
+    -------
+    real_time_series: :obj:`pd.DataFrame`
+        Real time series data in hourly resolution
+    """
+    if not type(nominal_time_series.index.year == pd.DatetimeIndex):
+        raise TypeError("Given time series must have a pd.DatetimeIndex!")
+
+    time_series_year = nominal_time_series.index.year[0]
+    real_time_series = nominal_time_series.div(
+        inflation_rate ** (time_series_year - year)
+    )
```

Comment:
Does it also make sense to include the check for integer values for the years?

Reply:
Hmm, in my view, this should be ensured by the user of the function and is stated in the docstring. 
Also, you can easily notice that from the error being thrown when you put in a non-numeric data type.

In principle, it is also possible to include explicit type checks in python. It is not so common, though, I think.

So something like
```python3

def my_function(integer_attr: int) -> float:
    """Function does fancy stuff with an integer input and returns a float"""
    return integer_attr + 0.2
```
is a thing you can do.